docs/book/stacks-and-components/component-guide/writing-custom-components.md

```diff
@@ -0,0 +1,239 @@
+            info: Info about the step that was executed.
+
+        Returns:
+            A dictionary of metadata.
+        """
+        return {}
+
+    def cleanup_step_run(self, info: "StepRunInfo", step_failed: bool) -> None:
+        """Cleans up resources after the step run is finished.
+
+        Args:
+            info: Info about the step that was executed.
+            step_failed: Whether the step failed.
+        """
+```
+
+See the full code [here](https://github.com/zenml-io/zenml/blob/main/src/zenml/stack/stack_component.py).
+
+{% hint style="info" %}
+
+In ZenML, a flavor's `config` and `settings` are two separate, yet related entities. The `config` is the static part of your flavor's configuration, defined when you register your flavor. The `settings` are the dynamic part of your flavor's configuration, optionally defined when you register your flavor but can be overridden at runtime. You can read more about the differences [here](../../user-guide/advanced-guide/pipelining-features/configure-steps-pipelines.md).
+
+{% endhint %}
+
+As each component might define a different interface, please take a look at each component type to read further details on how to define a custom component of that type. For example, read the [step operator](step-operators/custom.md) guide to learn how to create a custom step operator.
+
+## Registering a Flavor
+
+Once you have implemented your custom flavor and implementation class, you can register your component using the ZenML CLI. Use the `flavor register` command and provide the path to your flavor class:
+
+```shell
+# This points to the flavor class, not the implementation class
+zenml <component_type> flavor register <path.to.MyFlavor>
+```
+
+Replace `<component_type>` with the type of component you're developing the flavor for, and `<path.to.MyFlavor>` with the path to your flavor class.
+
+For example, if your custom orchestrator flavor class `MyOrchestratorFlavor` is defined in `flavors/my_flavor.py`, you'd register it by doing:
+
+```shell
+zenml orchestrator flavor register flavors.my_flavor.MyOrchestratorFlavor
+```
+
+### Listing Flavors
+
+You can list all registered flavors for a component type using the `flavor list` command:
+
+```shell
+zenml <component_type> flavor list
+```
+
+Replace `<component_type>` with the type of component you're interested in, e.g. `orchestrator`.
+
+### Understanding Flavor Registration
+
+When you register a flavor, ZenML imports the flavor class and stores its configuration in the ZenML repository. This allows ZenML to use your flavor when running pipelines, even if the dependencies required by the flavor are not installed in the current environment.
+
+In ZenML, the config is defined in flavor class in the `def config_class(self) -> Type[StackComponentConfig]` method, while in the settings are defined in the implementation class in the `def settings_class(self) -> Type[StackComponentSettings]` function.
+
+## Tips for Developing Flavors
```

Comment:
What I think is important here is to talk where the code is loaded from and how to make sure the whole team has access to it.

Reply:
I added one bullet, can you review @AlexejPenner

Reply:
Rewrote it a bit but looks good IMO so I'll resolve this